### DISCUSSION TITLE
Embedding libsql-server into an app

### DISCUSSION THREAD

**Discussion by lithdew** [tursodatabase/libsql#1783] [Ideas]
Hi,

I wanted to ask - has there been any work or attempts or plans on being able to embed libsql-server directly into an app?

For my use case, it would be great to be able to:
1. have my application's backend server host a SQLite3 instance in the same process address space for performance, and
2. remotely execute scripts to the SQLite3 instance, and
3. use Turso's CLI or via. any existing GUI tools supporting libsql-server's protocol to remotely manage the SQLite3 instance.

These 3 use cases are what's stopping me from using libsql (and in turn SQLite3) in production for my backend applications right now.

**Comment by avinassh** [comment]
Embedded replicas might solve your problem if you are fine with making writes to the server directly and app can have a read only instance: https://docs.turso.tech/features/embedded-replicas/introduction

May I offer you to join our discord discussion on Offline Writes? It is very close to what do you want. 

invite link: https://discord.gg/turso
channel: https://discord.com/channels/933071162680958986/1179836754891640882
